bookkeeper-server/src/main/java/org/apache/bookkeeper/discover/ZKRegistrationClient.java

```diff
@@ -327,9 +270,17 @@ public synchronized void unwatchWritableBookies(RegistrationListener listener) {
 
     @Override
     public synchronized CompletableFuture<Void> watchReadOnlyBookies(RegistrationListener listener) {
-        CompletableFuture<Void> f = new CompletableFuture<>();
+        CompletableFuture<Void> f;
```

Comment:
nit: add final to ensure all branches assign something. if returning a value is something miss from scala.